test/qa/lib/firewall.py

```diff
@@ -92,99 +97,219 @@
     "-A OUTPUT -d 10.0.0.0/8 -o eth0 -m comment --comment nordvpn -j ACCEPT",
 ]
 
+
+def _get_rules_killswitch_on(interface:str):
+    return \
+    [
+        f"-A INPUT -i {interface} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT",
+        f"-A INPUT -i {interface} -m comment --comment nordvpn -j DROP",
+        f"-A OUTPUT -o {interface} -m mark --mark 0xe1f1 -m comment --comment nordvpn -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff",
+        f"-A OUTPUT -o {interface} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT",
+        f"-A OUTPUT -o {interface} -m comment --comment nordvpn -j DROP"
+    ]
+
+
+def _get_rules_connected_to_vpn_server(interface:str):
+    return _get_rules_killswitch_on(interface)
+
+
+def _get_rules_allowlist_subnet_on(interface:str, subnets:list[str]):
+    # Subnet allowlist rules
+    result = []
+
+    for subnet in subnets:
+        result += f"-A INPUT -s {subnet} -i {interface} -m comment --comment nordvpn -j ACCEPT",
+
+    result += \
+    [
+        f"-A INPUT -i {interface} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT",
+        f"-A INPUT -i {interface} -m comment --comment nordvpn -j DROP",
+    ]
+
+    for subnet in subnets:
+        result += f"-A OUTPUT -d {subnet} -o {interface} -m comment --comment nordvpn -j ACCEPT",
+
+    result += \
+    [
+        f"-A OUTPUT -o {interface} -m mark --mark 0xe1f1 -m comment --comment nordvpn -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff",
+        f"-A OUTPUT -o {interface} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT",
+        f"-A OUTPUT -o {interface} -m comment --comment nordvpn -j DROP"
+    ]
+    return result
+
+
+def _get_rules_allowlist_port_on(interface:str, ports:list[lib.Port]):
+    # Port(range) whitelist rules for SPECIFIC protocols
+
+    ports_udp: list[lib.Port]
+    ports_tcp: list[lib.Port]
+    ports_udp, ports_tcp = _sort_ports_by_protocol(ports)
+
+    result = []
+
+    for port in ports_udp:
```

Comment:
This is the same as in function `_get_rules_allowlist_subnet_and_port_on`. Please move this to a separate function and re-use that. Also, check if the same can be done with other sets of FW rules.